freqtrade/configuration/config_validation.py

```diff
@@ -334,6 +335,26 @@ def _validate_freqai_hyperopt(conf: Dict[str, Any]) -> None:
             'Using analyze-per-epoch parameter is not supported with a FreqAI strategy.')
 
 
+def _validate_freqai_backtest(conf: Dict[str, Any]) -> None:
+    if conf.get('runmode', RunMode.OTHER) == RunMode.BACKTEST:
+        freqai_enabled = conf.get('freqai', {}).get('enabled', False)
+        timerange = conf.get('timerange')
+        freqai_backtest_live_models = conf.get('freqai_backtest_live_models', False)
+        if freqai_backtest_live_models and freqai_enabled and timerange:
+            raise OperationalException(
+                'Using timerange parameter is not supported with '
+                '--freqai-backtest-live-models parameter.')
+
+        if freqai_backtest_live_models and not freqai_enabled:
+            raise OperationalException(
+                'Using --freqai-backtest-live-models parameter is only '
+                'supported with a FreqAI strategy.')
+
+        if freqai_enabled and not freqai_backtest_live_models and not timerange:
+            raise OperationalException(
+                'Please pass --timerange if you intend to use FreqAI for backtesting.')
```

Comment:
huh now that's some extensive logic as to when we can and cannot do something.

I'm not sure users will understand this...

Reply:
The timerange check was inside `freqtrade.freqai.data_kitchen` file. I moved to this file, to keep all validations together. In summary.

- Timerange is required for a regular FreqAI backtest.
- Backtest live models will calculate timerange, and we don't need this param.
- Backtest live models only works with FreqAI

I agree that we need to document well the rules.

Reply:
i'm not questioning the logic / need (even if it wouldn't have been in data_kitchen - doing sanity checks is still fine.
it's just a lot of "this combination, not that combination" :laughing: and without docs - it takes a moment to grasp :laughing: